reporting/reporting.go

```diff
@@ -0,0 +1,127 @@
+package reporting
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/honeycombio/libhoney-go"
+
+	"github.com/honeycombio/honeytail/event"
+)
+
+const (
+	contextKeyBuilder = "reporting"
+	reportingSuffix   = "-reports"
+)
+
+func NewContext(ctx context.Context) context.Context {
+	builder := libhoney.NewBuilder()
+	return context.WithValue(ctx, contextKeyBuilder, builder)
+}
+
+func Options(ctx context.Context, options interface{}) {
+	if ev := getEvent(ctx); ev != nil {
+		optj, err := json.Marshal(options)
+		if err != nil {
+			return
+		}
+		ev.AddField("config_json", string(optj))
+		ev.Send()
+	}
+}
+
+func TailState(ctx context.Context, file string, inode uint64, offset int64, size int64) {
+	if ev := getEvent(ctx); ev != nil {
+		ev.AddField("tail_filename", file)
+		ev.AddField("tail_inode", inode)
+		ev.AddField("tail_offset", offset)
+		ev.AddField("tail_filesize", size)
+		ev.AddField("pct_processed", float64(offset)/float64(size)*100)
+		ev.Send()
+	}
+}
+
+func ParseError(ctx context.Context, line string, err error) {
+	logrus.WithFields(logrus.Fields{
+		"line":  line,
+		"error": err,
+	}).Debugln("Skipped: log line failed to parse")
+
+	if ev := getEvent(ctx); ev != nil {
+		ev.AddField("log_line", line)
+		ev.AddField("parse_error", err.Error())
+		ev.Send()
+	}
+}
+
+func Skip(ctx context.Context, line, skipReason string) {
+	SkipWithFields(ctx, line, skipReason, nil)
+}
+
+func SkipWithFields(ctx context.Context, line, skipReason string, fields logrus.Fields) {
+	logrus.WithFields(fields).WithField("line", line).Debugln("Skipped:", skipReason)
+
+	if ev := getEvent(ctx); ev != nil {
+		ev.AddField("log_line", line)
+		ev.AddField("skip_reason", skipReason)
+		ev.Send()
```

Comment:
could we add `ev.Add(fields)` too?

Reply:
I turned up my nose at the `fields` because they a) either repeat data that already exists (e.g. `line` and `rawE`, or `samplerate` in the config) or b) are... redundant (e.g. the value of `matched`, which can be otherwise determined by the value of `InvertFilter` and this log line being output.

But I see your hoping for consistency. Would you be horrified if I instead axed those additional fields from those log lines? 😶